README.md

```diff
@@ -11,572 +11,528 @@
-"blazing fast" for streaming libraries, it competes with lists and `vector`.
-Other streaming libraries like "streaming", "pipes" and "conduit" are orders of
-magnitude slower on most microbenchmarks.  See [streaming
-benchmarks](https://github.com/composewell/streaming-benchmarks) for detailed
-comparison.
-
-The following chart shows a comparison of those streamly and list operations
-where performance of the two differs by more than 10%. Positive y-axis displays
-how many times worse is a list operation compared to the same streamly
-operation, negative y-axis shows where streamly is worse compared to lists.
+## Idiomatic Haskell at the speed of C
 
-![Streamly vs Lists (time) comparison](charts-0/streamly-vs-list-time.svg)
+Streamly is a Haskell library/framework providing basic building blocks
+or combinators to perform fundamental as well as advanced programming
+tasks with ease. The key features it provides are:
 
-Streamly uses lock-free synchronization for concurrent operations. It employs
-auto-scaling of the degree of concurrency based on demand. For CPU bound tasks
-it tries to keep the threads close to the number of CPUs available whereas for
-IO bound tasks more threads can be utilized. Parallelism can be utilized with
-little overhead even if the task size is very small.  See [concurrency
-benchmarks](https://github.com/composewell/concurrency-benchmarks) for detailed
-performance results and a comparison with the `async` package.
+* Speed of C
+* Safety of Haskell
+* Idiomatic functional programming
+* Powerful abstractions
+* Declarative concurrency
 
-## Installing and using
+Let's go through some practical examples to see it working. You
+can find the working code of these examples in the [streamly-examples
+repository](https://github.com/composewell/streamly-examples).
 
-Please see [INSTALL.md](./INSTALL.md) for instructions on how to use streamly
-with your Haskell build tool or package manager. You may want to go through it
-before jumping to run the examples below.
+## Types Overview
 
-The module `Streamly.Prelude` provides the core stream types and combinators
-for type casting, controlling concurrency, stream construction, transformation,
-folding, merging, and zipping.
+* `SerialT IO a` is a serial stream of values of type `a` in IO Monad.
+* `AsyncT IO a` is a concurrent (async) stream of values of type `a` in IO
+  Monad.
+* `Unfold IO a b` is a representation of a function that converts a seed
+  value of type `a` to a stream of values of type `b` in IO Monad.
+* `Fold IO a b` is a representation of a function that converts a stream of
+  type `a` to a final accumulator of type `b` in IO Monad.
 
-## Streaming Pipelines
+## Modular Word Counting
 
-The following snippet provides a simple stream composition example that reads
-numbers from stdin, prints the squares of even numbers and exits if an even
-number more than 9 is entered.
+The `Fold` data type in streamly represents a consumer of stream. In
```

Comment:
Is this the correct place to put info about the `Fold` data type?

Reply:
We are talking about composing folds in this para, so a brief intro of fold is needed.